spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/MilliSpan.java

```diff
@@ -66,9 +66,10 @@ public MilliSpan(long begin, long end, String name, String traceId, List<String>
 	private MilliSpan() {
 		this.begin = 0;
 		this.name = null;
-		this.traceId = null;
+		this.traceId = 0;
```

Comment:
really uncomfortable about this.. maybe add a TODO, as it looks like one could accidentally end up with trace id 0, which is later documented as random.